### PR TITLE
Support aeson-2.0

### LINKED ISSUE #680: aeson ^>= 2.0 support
We deferred this so we didn't have to worry about it during the 2.0 stabilisation, but we definitely want to keep up with it once we get our 2.0 out the door.

### PR DESCRIPTION
**Pull request by endgame**
Closes #680 .

Tests done:

- [X] Built repo with `constraints: aeson ^>= 2.0`
- [X] Built repo with `constraints: aeson ^>= 1.5`
- [x] Write a test program and ensure HashedBody still works

### CODE REVIEW

File: configs/services/sagemaker-runtime.json [hunk 1]
```diff
@@ -5,9 +5,7 @@
             "replacedBy": {
                 "name": "Prelude.Text",
                 "underive": [
-                    "show",
-                    "read",
-                    "eq"
+                    "read"
```
Comment (endgame) [inline configs/services/sagemaker-runtime.json:8]: New `hashable` has `Eq` as superclass.

**Review by endgame — COMMENTED** [review]
`sagemaker-runtime` will need regenerating but you prefer to do those in other branches.

**Comment by endgame** [comment]
I banged together a test program that successfully uploaded a HashedBody to S3, so I'm pretty sure I didn't break anything. @brendanhay OK to merge?

**Comment by brendanhay** [comment]
LGTM

**Pull request merged**